rs/index/src/ivf/builder.rs

```diff
@@ -37,22 +37,31 @@ impl IvfBuilder {
         let _ = create_dir(vectors_path.clone());
```

Comment:
you'll need to handle the case where we fail to create directory for any reason. Maybe return `Result<Self>` in this function?